sympy/core/evalf.py

```diff
@@ -187,13 +223,13 @@ def scaled_zero(mag, sign=1):
         raise ValueError('scaled zero expects int or scaled_zero tuple.')
 
 
-def iszero(mpf, scaled=False):
+def iszero(mpf: tUnion[MPF_TUP, SCALED_ZERO_TUP, None], scaled=False) -> Optional[bool]:
     if not scaled:
         return not mpf or not mpf[1] and not mpf[-1]
     return mpf and isinstance(mpf[0], list) and mpf[1] == mpf[-1] == 1
 
 
-def complex_accuracy(result):
+def complex_accuracy(result: TMP_RES) -> tUnion[int, Any]:
```

Comment:
Is the return type here not `tUnion[int, float]`?

Reply:
https://github.com/sympy/sympy/blob/dfd966437a2ac0551c2c55174187f92dd6cba278/sympy/core/evalf.py#L557-L566
float will cause a float value to be assigned to int `prec`

Reply:
So what does happen in this code path if `complex_accuracy` returns `INF`?

Reply:
`target_prec - acc` is minus inf then, so the `max` returns `10 + 2**i` and `prec` remains int.